packages/remix-react/components.tsx

```diff
@@ -1298,6 +1293,182 @@ export function useSubmitImpl(key?: string): SubmitFunction {
+  //        - The field element is an input element whose type attribute is in the Radio Button state and whose checkedness is false.
+  //       Then continue.
+  //
+  // IMPLEMENTATION NOTES:
+  //  - We skip the datalist check since 1. browsers don't do it and 2. the only valid child of a datalist
+  //    is an option, which is not a submittable element.
+  //  - The "button is not submitter" logic is handled at the start of the for loop
+  let disabled = ":disabled";
+  let uncheckedCheckboxOrRadio = "input[type~='radio checkbox']:not([checked])";
+  let unNamedExcludingImageButtons =
+    ":is([name=''],:not([name]):not(input[type=image])";
+  let disqualifiers = [
+    disabled,
+    uncheckedCheckboxOrRadio,
+    unNamedExcludingImageButtons,
+  ].join(",");
+
+  let selector = `${submittable}${ownedByForm}:not(${disqualifiers})`;
+  let controls = scope.querySelectorAll<SubmittableElement>(selector);
+
+  for (let field of controls) {
+    // IMPLEMENTATION NOTES:
+    //  - "button" is defined in the spec as any element with the prose "The element is a button"
+    //    See: https://html.spec.whatwg.org/multipage/input.html
+    //     and https://html.spec.whatwg.org/multipage/form-elements.html
+    let isButton =
+      isButtonElement(field) ||
+      (isInputElement(field) &&
+        ["submit", "image", "button", "reset"].includes(field.type));
+    if (isButton && field !== submitter) continue;
+
+    //  2. If the field element is an input element whose type attribute is in the Image Button state, then:
+    //     1. If the field element has a name attribute specified and its value is not the empty string, let name be that value followed by a single U+002E FULL STOP character (.). Otherwise, let name be the empty string.
+    //     2. Let namex be the string consisting of the concatenation of name and a single U+0078 LATIN SMALL LETTER X character (x).
+    //     3. Let namey be the string consisting of the concatenation of name and a single U+0079 LATIN SMALL LETTER Y character (y).
+    //     4. The field element is submitter, and before this algorithm was invoked the user indicated a coordinate. Let x be the x-component of the coordinate selected by the user, and let y be the y-component of the coordinate selected by the user.
+    //     5. Create an entry with namex and x, and append it to entry list.
+    //     6. Create an entry with namey and y, and append it to entry list.
+    //     7. Continue.
+    if (
+      isImageButtonElement(field) &&
+      field === submitter &&
+      field[SELECTED_COORDINATE]
+    ) {
+      let prefix = field.name ? `${field.name}.` : "";
+      let { x, y } = field[SELECTED_COORDINATE];
+      entryList.push([`${prefix}x`, String(x)]);
+      entryList.push([`${prefix}y`, String(y)]);
+      continue;
+    }
+
+    //  3. If the field is a form-associated custom element, then perform the entry construction algorithm given field and entry list, then continue.
```

Comment:
oops, forgot about this line... might need to see if this is even doable, otherwise add an implementation note here

Reply:
yeah this doesn't seem doable, added a comment here with more context